diff.py

```diff
@@ -1495,12 +1495,10 @@ def dump_objfile(
 def dump_binary(
     start: str, end: Optional[str], config: Config, project: ProjectSettings
 ) -> Tuple[str, ObjdumpCommand, ObjdumpCommand]:
-    if not project.baseimg or not project.myimg and not config.file:
+    binfile = config.file or project.myimg
+    if not project.baseimg or not binfile:
         fail("Missing myimg/baseimg in config.")
-    binfile = project.myimg
-    if config.file is not None:
-        binfile = config.file
-    if config.make:
+    if config.make and project.myimg is not None:
         run_make(project.myimg, project)
```

Comment:
hm, doesn't it make more sense to run make on `binfile` btw?

Reply:
My thoughts were that the file specified on the command line would be updated/produced as a byproduct of running make on `myimg` as `binfile` likely isn't a make target.

I suppose my thought process was skewed by my specific use case (which doesn't use `-m`) with one build command that outputs a ton of files. It would be more logical to try to run `make binfile` instead.